model/task/db.go

```diff
@@ -963,6 +963,10 @@ func FindTaskNamesByBuildVariant(projectId string, buildVariant string) ([]strin
 				{DisplayTaskIdKey: bson.M{"$exists": false}},
 				{DisplayTaskIdKey: ""},
 			},
+			"$and": []bson.M{
+				{RevisionOrderNumberKey: bson.M{"$gte": repoOrderNumber - VersionLimit}},
```

Comment:
I think technically this should just be $gt to respect the version limit.

Reply:
done.